spring-boot-project/spring-boot/src/main/java/org/springframework/boot/info/ProcessInfo.java

```diff
@@ -72,6 +74,34 @@ public MemoryInfo getMemory() {
 		return new MemoryInfo();
 	}
 
+	/**
+	 * Virtual threads information for the process. These values provide details about the
+	 * current state of virtual threads, including the number of mounted threads, queued threads,
+	 * the parallelism level, and the thread pool size.
+	 *
+	 * @return an instance of {@link VirtualThreadsInfo} containing information about virtual threads,
+	 * or {@code null} if the VirtualThreadSchedulerMXBean is not available.
+	 * @since 3.4.2
+	 */
+	public VirtualThreadsInfo getVirtualThreads() {
+		try {
+			@SuppressWarnings("unchecked")
+			Class<? extends PlatformManagedObject> mxBeanClass =
+					(Class<? extends PlatformManagedObject>) Class.forName("jdk.management.VirtualThreadSchedulerMXBean");
+
```

Comment:
Please remove the whitespace. We generally don't have any in the method body. If things get too crowded we treat that as a sign to do an "extract method" refactor.